Client/Frontend/Settings/Brave Today/BraveTodayAddSourceViewController.swift

```diff
@@ -0,0 +1,392 @@
+        configuration.timeoutIntervalForRequest = 5
+        return URLSession(configuration: configuration, delegate: nil, delegateQueue: .main)
+    }()
+    
+    private func displayError(_ error: FindFeedsError) {
+        let alert = UIAlertController(title: Strings.BraveToday.addSourceFailureTitle, message: error.localizedDescription, preferredStyle: .alert)
+        alert.addAction(.init(title: Strings.OKString, style: .default, handler: nil))
+        present(alert, animated: true)
+    }
+    
+    private func searchPageForFeeds() {
+        guard var text = textField.text else { return }
+        if text.hasPrefix("feed:"), let range = text.range(of: "feed:") {
+            text.replaceSubrange(range, with: [])
+        }
+        guard let url = URIFixup.getURL(text) else { return }
+        isLoading = true
+        downloadPageData(for: url) { [weak self] result in
+            guard let self = self else { return }
+            self.isLoading = false
+            switch result {
+            case .success(let data):
+                let resultsController = BraveTodayAddSourceResultsViewController(
+                    dataSource: self.feedDataSource,
+                    searchedURL: url,
+                    rssFeedLocations: data,
+                    sourcesAdded: self.sourcesAdded
+                )
+                self.navigationController?.pushViewController(resultsController, animated: true)
+            case .failure(let error):
+                self.displayError(error)
+            }
+        }
+    }
+    
+    private enum FindFeedsError: Error {
+        /// An error occured while attempting to download the page
+        case dataTaskError(Error)
+        /// The data was either not received or is in the incorrect format
+        case invalidData
+        /// The data downloaded did not match a
+        case parserError(ParserError)
+        /// No feeds were found at the given URL
+        case noFeedsFound
+        
+        var localizedDescription: String {
+            switch self {
+            case .dataTaskError(let error as URLError) where error.code == .notConnectedToInternet:
+                return error.localizedDescription
+            case .dataTaskError:
+                return Strings.BraveToday.addSourceNetworkFailureMessage
+            case .invalidData, .parserError:
+                return Strings.BraveToday.addSourceInvalidDataMessage
+            case .noFeedsFound:
+                return Strings.BraveToday.addSourceNoFeedsFoundMessage
+            }
+        }
+    }
+    
+    private var pageTask: URLSessionDataTask?
+    private func downloadPageData(for url: URL, _ completion: @escaping (Result<[RSSFeedLocation], FindFeedsError>) -> Void) {
+        pageTask = session.dataTask(with: url) { [weak self] (data, response, error) in
+            guard let self = self else { return }
+            if let error = error as? URLError, error.code == .cancelled {
+                return
```

Comment:
do we not want to return completion handler here?

Reply:
Good point, no reason to handle cancelled and return without handler. If we dismiss the screen, the error will not display anyways since `self` will be `nil`